### PR TITLE
Makefile cleanup - Improved handling of different TARGETS, removed duplication.

### PR DESCRIPTION
**Pull request by martinbudden**
@hydra, you probably want to take this now. A small step in improving the maintainability of the `Makefile`.

### CODE REVIEW

**Comment by ledvinap** [comment]
Nice work

Another possibility is to include target-specific makefile into main and define things like memory sizes and supported features there .. But maybe it will make things more complicated ..

File: Makefile [hunk 1]
```diff
@@ -35,18 +35,28 @@ FLASH_SIZE ?=
...
-VALID_TARGETS	 = ALIENWIIF1 ALIENWIIF3 CC3D CHEBUZZF3 CJMCU COLIBRI_RACE EUSTM32F103RC LUX_RACE MOTOLAB NAZE NAZE32PRO OLIMEXINO PORT103R RMDO SPARKY SPRACINGF3 SPRACINGF3EVO SPRACINGF3MINI STM32F3DISCOVERY IRCFUSIONF3 
+64K_TARGETS  = CJMCU
+128K_TARGETS = ALIENWIIF1 CC3D NAZE OLIMEXINO RMDO
+256K_TARGETS = ALIENWIIF3 CHEBUZZF3 COLIBRI_RACE EUSTM32F103RC IRCFUSIONF3 LUX_RACE MOTOLAB NAZE32PRO PORT103R SPARKY SPRACINGF3 SPRACINGF3EVO SPRACINGF3MINI STM32F3DISCOVERY
+
+F3_TARGETS = ALIENWIIF3 CHEBUZZF3 COLIBRI_RACE IRCFUSIONF3 LUX_RACE MOTOLAB NAZE32PRO RMDO SPARKY SPRACINGF3 SPRACINGF3EVO SPRACINGF3MINI STM32F3DISCOVERY
+
+VALID_TARGETS = $(64K_TARGETS) $(128K_TARGETS) $(256K_TARGETS)
+
+# Valid targets for OP VCP support
```
Comment (hydra) [inline Makefile]: OP?

File: Makefile [hunk 2]
```diff
@@ -35,18 +35,28 @@ FLASH_SIZE ?=
...
+VALID_VCP_TARGETS = CC3D IRCFUSIONF3 SPRACINGF3
```
Comment (hydra) [inline Makefile]: this doesn't make sense.  the SPRACINGF3 does NOT have VCP.

**Comment by hydra** [comment]
since this is a makefile cleanup, we should defer this to 1.14.0 - code cleanups yes, now.  makefile can wait a little bit.

File: Makefile [hunk 3]
```diff
@@ -35,18 +35,27 @@ FLASH_SIZE ?=
...
+64K_TARGETS  = CJMCU
+128K_TARGETS = ALIENFLIGHTF1 CC3D NAZE OLIMEXINO RMDO
+256K_TARGETS = ALIENFLIGHTF3 CHEBUZZF3 COLIBRI_RACE EUSTM32F103RC IRCFUSIONF3 LUX_RACE MOTOLAB NAZE32PRO PORT103R SPARKY SPRACINGF3 SPRACINGF3EVO SPRACINGF3MINI STM32F3DISCOVERY
+
+F3_TARGETS = ALIENFLIGHTF3 CHEBUZZF3 COLIBRI_RACE IRCFUSIONF3 LUX_RACE MOTOLAB NAZE32PRO RMDO SPARKY SPRACINGF3 SPRACINGF3EVO SPRACINGF3MINI STM32F3DISCOVERY
+
+VALID_TARGETS = $(64K_TARGETS) $(128K_TARGETS) $(256K_TARGETS)
+
+VCP_TARGETS = CC3D ALIENFLIGHTF3 CHEBUZZF3 COLIBRI_RACE LUX_RACE MOTOLAB NAZE32PRO SPARKY SPRACINGF3EVO SPRACINGF3MINI STM32F3DISCOVERY
+FATFS_TARGETS = SPRACINGF3MINI
```
Comment (hydra) [inline Makefile]: SPRACINGF3EVO too

**Comment by hydra** [comment]
I'll give this a whirl here and merge this for the final release of 1.13.0 since I need to get started on some other Makefile related changes.

File: Makefile (hunk 3 above)
Comment (hydra) [inline Makefile]: haha, lol, except that was referring to the old fatfs code which isn't used.  i will clean that up.

**Pull request merged**